avm/ptn/azd/ml-project/main.bicep

```diff
@@ -0,0 +1,171 @@
+metadata name = 'Machine Learning workspace'
+metadata description = '''Create a machine learning workspace, configure the key vault access policy and assign role permissions to the machine learning instance.
+
+**Note:** This module is not intended for broad, generic use, as it was designed to cater for the requirements of the AZD CLI product. Feature requests and bug fix requests are welcome if they support the development of the AZD CLI but may not be incorporated if they aim to make this module more generic than what it needs to be for its primary use case.'''
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. The name of the machine learning workspace.')
+param name string
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Tags of the resource.')
+@metadata({
+  example: '''
+  {
+      "key1": "value1"
+      "key2": "value2"
+  }
+  '''
+})
+param tags object?
+
+@description('Optional. Specifies the SKU, also referred as \'edition\' of the Azure Machine Learning workspace.')
+@allowed([
+  'Free'
+  'Basic'
+  'Standard'
+  'Premium'
+])
+param projectSku string = 'Basic'
+
+@description('Optional. The type of Azure Machine Learning workspace to create.')
+@allowed([
+  'Default'
+  'Project'
+  'Hub'
+  'FeatureStore'
+])
+param projectKind string = 'Project'
+
+@description('Optional. The flag to signal HBI data in the workspace and reduce diagnostic data collected by the service.')
+param hbiWorkspace bool = false
+
+@description('Optional. Whether or not public network access is allowed for this machine learning resource. For security reasons it should be disabled.')
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+param publicNetworkAccess string = 'Enabled'
```

Comment:
Is this intended to be enabled by default?

Reply:
Yes, it is aligned with the default values ​​of the [project.bicep](https://github.com/Azure/azure-dev/blob/main/templates/common/infra/bicep/core/ai/project.bicep#L17) file parameters in infra/core.